tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/trajopt/config/trajopt_planner_config.h

```diff
@@ -63,6 +63,13 @@ struct TrajOptPlannerConfig
   /** @brief Trajopt problem to be solved (Required) */
   trajopt::TrajOptProb::Ptr prob;
 
+  /**@brief Special link collision cost distances
+   * This is a vector of std::tuple<link1, link2, collision distance, collision coefficient */
+  std::vector<std::tuple<std::string, std::string, double, double>> special_collision_cost;
```

Comment:
Why not use the Trajopt SafetyMarginData class here?

Reply:
I don't fully understand how those work. In order to use them would I have to have both a SafetyMarginData struct and a vector of pairs of strings for both the cost and collision terms so that I can call getPairSafetyMarginData later when I want to modify the default collision terms being created in trajopt_planner_default_config.cpp? I am unaware of a way to iterate through the SafetyMarginData class, but I'm all for using something already created rather than a vector of an arbitrary tuple like I have defined here.

Reply:
> I don't fully understand how those work. In order to use them would I have to have both a SafetyMarginData struct and a vector of pairs of strings for both the cost and collision terms so that I can call getPairSafetyMarginData later when I want to modify the default collision terms being created in trajopt_planner_default_config.cpp?

I recommend taking a look at the class which is documented. It stores the pair internally and if you want to overwrite an existing one you just setPairSafetyMarginData again. Currently there is no way to remove one, but I would add a removePairSafetyMarginData function to the class. 

> I am unaware of a way to iterate through the SafetyMarginData class, but I'm all for using something already created rather than a vector of an arbitrary tuple like I have defined here.

In most case you iterate over the contacts and query getPairSafetyMarginData to check if it is a valid contact.

Reply:
Okay, I was able to get SafetyMarginData classes to work for me